### PR TITLE
Docs, cleanup, examples, and simplified config

### PR DESCRIPTION
**Pull request by HenrikJoreteg**
major improvement to docs
unified configuration
adding ability to pass html to defaultTemplate
now including babel proper by default
add warning/prep for peerDependency deprecation

would love some feedback on this from @lukekarrys and @latentflip <3

### CODE REVIEW

File: lib/html-plugin.js [hunk 1]
```diff
@@ -39,8 +56,14 @@ HJSPlugin.prototype.apply = function (compiler) {
     // this is useful if you want default
     // html but want to create multiple or
     // differently named HTML files
-    assets.defaultTemplate = function () {
-      return defaultHtml(assets)
+    assets.defaultTemplate = function (opts) {
+      var templateData = assign({
+        charset: 'utf-8',
+        metaViewPort: true,
```
Comment (lukekarrys) [inline lib/html-plugin.js]: Should this option be `metaViewport`?

**Comment by lukekarrys** [comment]
Seems pretty good to me visually. The one line note I left is a small thing, and probably doesnt matter if thats not a public option since I didn't see it documented anywhere else. But if the user is allowed to set it, I think it should be changed.

Also @HenrikJoreteg, I assume this is `v2`? If so, could we hold off on publishing until we see how the other PRs fit into this?

File: lib/html-plugin.js (hunk 1 above)
Comment (HenrikJoreteg) [inline lib/html-plugin.js]: will fix

File: README.md [hunk 2]
```diff
@@ -50,27 +75,183 @@ module.exports = getConfig({
...
-    "build": "NODE_ENV=production webpack",
-    "deploy": "npm run build && surge -p public"
-  },
-  "dependencies": {
-    "webpack": "*",
-    "hjs-webpack": "*",
-    "surge": "*"
+**using an `html` function to generate specific files**
+
+This is where it gets interesting. Imagine pre-rendering all known structural content for a Native Web App to static files. Users get pixels on the screen immediately, your JS takes over when downloaded. If you're using React, this "taking over" can be completely seemless and invisible to the user. It's also possible with this approach to write an app that works entirely without JS. See the [prerendered-html-files example](https://github.com/HenrikJoreteg/hjs-webpack/tree/master/examples/prerendered-html-files).
```
Comment (aaronmccall) [inline README.md]: s/seemless/seamless in "If you're using React, this "taking over" can be completely seemless and invisible to the user."

File: bin/hjs-dev-server [hunk 3]
```diff
@@ -9,21 +9,24 @@ var rootFolder = findRoot(process.env.PWD)
 
 // get config
 if (argv.config) {
+
+    console.log('config', argv.config)
```
Comment (latentflip) [inline bin/hjs-dev-server]: This and the one below probably didn't mean to get committed.

**Comment by latentflip** [comment]
Looks good, but I think we should remove hjs-dev-server. Webpack already allows you to specify devserver config from webpack.config.js (http://webpack.github.io/docs/webpack-dev-server.html#cli). I think hjs-wepback should just prefill those (hot/history-api) and extend any extra options you pass in, and use webpack-dev-server as a peer dep.

That way I can toggle that config from the cli/webpack config as I see fit, and not have to wonder what hjs-dev-server is doing.

**Comment by HenrikJoreteg** [comment]
added changelog: https://github.com/henrikjoreteg/hjs-webpack#changelog and published as `2.0.0` thanks all!

**Pull request merged**